Source/Private/LevelStatsCollector.cpp

```diff
@@ -1,24 +1,65 @@
 ﻿#include "LevelStatsCollector.h"
 
+#include "LevelStatsCollectorState.h"
+
 #include <Components/SceneCaptureComponent2D.h>
+#include <Engine/Engine.h>
 #include <Engine/LevelBounds.h>
 #include <Engine/TextureRenderTarget2D.h>
 #include <ImageUtils.h>
 
+FCustomPerformanceChart::FCustomPerformanceChart( const FDateTime & in_start_time, const FString & in_chart_label, const FString & in_output_path ) :
+    FPerformanceTrackingChart( in_start_time, in_chart_label ),
+    CustomOutputPath( in_output_path )
+{
+}
+
+void FCustomPerformanceChart::DumpFPSChartToCustomLocation( const FString & in_map_name )
+{
+    TArray< const FPerformanceTrackingChart * > charts;
+    charts.Add( this );
+
+    DumpChartsToOutputLog( AccumulatedChartTime, charts, in_map_name );
+
+#if ALLOW_DEBUG_FILES
+    IFileManager::Get().MakeDirectory( *CustomOutputPath, true );
+
+    {
+        const auto log_filename = CustomOutputPath / CreateFileNameForChart( TEXT( "FPS" ), in_map_name, TEXT( ".log" ) );
+        DumpChartsToLogFile( AccumulatedChartTime, charts, in_map_name, log_filename );
+    }
+
+    {
+        const auto map_and_chart_label = ChartLabel.IsEmpty() ? in_map_name : ( ChartLabel + TEXT( "-" ) + in_map_name );
+        const auto html_filename = CustomOutputPath / CreateFileNameForChart( TEXT( "FPS" ),
+                                                          *( map_and_chart_label + TEXT( "-" ) + CaptureStartTime.ToString() ),
+                                                          TEXT( ".html" ) );
+        DumpChartsToHTML( AccumulatedChartTime, charts, map_and_chart_label, html_filename );
+    }
+#endif
+}
+
+FString FCustomPerformanceChart::CreateFileNameForChart( const FString & /* chart_type */, const FString & /* in_map_name */, const FString & file_extension )
+{
+    const FString platform = FPlatformProperties::PlatformName();
+    return TEXT( "metrics" ) + file_extension;
+}
+
 ALevelStatsCollector::ALevelStatsCollector() :
-    CellSize( 10000.0f ),
     GridCenterOffset( FVector::ZeroVector ),
+    TotalCaptureCount( 0 ),
+    CurrentCellIndex( 0 ),
+    MetricsDuration( 5.0f ),
+    MetricsWaitDelay( 1.0f ),
+    CellSize( 10000.0f ),
     CameraHeight( 10000.0f ),
     CameraHeightOffset( 250.0f ),
     CameraRotationDelta( 90.0f ),
     CaptureDelay( 0.1f ),
-    OutputDirectory( TEXT( "Saved/Screenshots/LevelStatsCollector/" ) ),
-    CurrentCellIndex( 0 ),
     CurrentRotation( 0.0f ),
     CurrentCaptureDelay( 0.0f ),
-    TotalCaptureCount( 0 ),
-    bIsCapturing( false ),
-    bIsInitialized( false )
+    IsCaptureInProgress( false ),
```

Comment:
bIs

Reply:
coding standards says we should not use bIs but Subject+Verb+Adjective form